src/distilabel/steps/task/evol_instruct/base.py

```diff
@@ -0,0 +1,213 @@
+        """Override this method to perform additional initialization after `__init__` and `model_construct`.
+        This is useful if you want to do some validation that requires the entire model to be initialized.
+        """
+        super().model_post_init(__context)
+
+        np.random.seed(42)
+
+    @cached_property
+    def _english_nouns(self) -> List[str]:
+        """A list of English nouns to be used as part of the starting prompts for the task.
+
+        Reference:
+            - https://github.com/h2oai/h2o-wizardlm
+        """
+        _path = str(
+            importlib_resources.files("distilabel")
+            / "steps/task/evol_instruct/english_nouns.txt"
+        )
+        with open(_path, mode="r") as f:
+            return [line.strip() for line in f.readlines()]
+
+    @property
+    def inputs(self) -> List[str]:
+        """The input for the task is the `instruction`."""
+        return ["instruction"]
+
+    def format_input(self, input: str) -> ChatType:  # type: ignore
+        """The input is formatted as a `ChatType` assuming that the instruction
+        is the first interaction from the user within a conversation. And the
+        `system_prompt` is added as the first message if it exists."""
+        return [{"role": "user", "content": input}]
+
+    @property
+    def outputs(self) -> List[str]:
+        """The output for the task is the `generation` and the `model_name`."""
+        # TODO: having to define a `model_name` column every time as the `Task.outputs` is not ideal,
+        # this could be handled always and the value could be included within the DAG validation when
+        # a `Task` is used, since all the `Task` subclasses will have an `llm` with a `model_name` attr.
+        _outputs = [
+            "evolved_instruction"
+            if not self.store_evolutions
+            else "evolved_instructions",
+            "model_name",
+        ]
+        if self.generate_answers:
+            _outputs.append("answer")
+        return _outputs
+
+    def format_output(
+        self, instructions: Union[str, List[str]], answer: Optional[str] = None
+    ) -> Dict[str, Any]:  # type: ignore
+        """The output is formatted as a dictionary with the `instruction`. The `model_name`
+        will be automatically included."""
+        if not self.output_mappings:
+            self.output_mappings = {k: k for k in self.outputs}
+
+        _output = {}
+        if not self.store_evolutions:
+            _output[self.output_mappings["evolved_instruction"]] = instructions[-1]
+        else:
+            _output[self.output_mappings["evolved_instructions"]] = instructions
+
+        if self.generate_answers and answer is not None:
+            _output[self.output_mappings["answer"]] = answer
+
+        _output[self.output_mappings["model_name"]] = self.llm.model_name
+        return _output
+
+    @override
+    def process(self, inputs: "StepInput") -> "StepOutput":  # type: ignore
```

Comment:
I think we could split this method in two major methods, that then are used in `process`: `_evolve_instructions` and `_generate_answers`